scripts/utils/publishNewVersion.js

```diff
@@ -51,22 +51,18 @@ function publishToNpm(npmTag) {
 }
 
 function releaseDocsVersion(newVersion, previousVersion) {
-  const shouldReplaceWithPreviousVersion = previousVersion && isSameMajorVersion(newVersion, previousVersion);
-  if (shouldReplaceWithPreviousVersion) {
-    removeDocsForVersion(previousVersion);
+  const newDocsVersion = newVersion.split('.', 1).concat('.x');
+  const previousDocsVersion = previousVersion.split('.', 1).concat('.x');
+
+  if (newDocsVersion === previousDocsVersion) {
+    removeDocsForVersion(previousDocsVersion);
   }
 
-  buildDocsForVersion(newVersion);
+  buildDocsForVersion(newDocsVersion);
```

Comment:
If this fails and `removeDocsForVersion` we are left with no version? 😿

Reply:
We are left without the previous docs version (that we just removed), but why will this happen?
We can manually release the docs version in this scenario.

Reply:
Also, the replacement has to happen in this order (remove -> add) since we replace one in another (doc versions) while they share the same name.

Reply:
ok, as long as there's a fallback